packages/hardhat/hardhat.config.ts

```diff
@@ -38,116 +27,39 @@ const config: HardhatUserConfig = {
       },
     ],
   },
-  defaultNetwork: "localhost",
-  namedAccounts: {
-    deployer: {
-      // By default, it will take the first Hardhat account as the deployer
-      default: 0,
-    },
-  },
   networks: {
-    // View the networks that are pre-configured.
-    // If the network you are looking for is not here you can add new network settings
-    hardhat: {
-      forking: {
-        url: `https://eth-mainnet.alchemyapi.io/v2/${providerApiKey}`,
-        enabled: process.env.MAINNET_FORKING_ENABLED === "true",
-      },
-    },
     mainnet: {
+      type: "http",
+      chainType: "l1",
       url: "https://mainnet.rpc.buidlguidl.com",
-      accounts: [deployerPrivateKey],
+      accounts: [configVariable("DEPLOYER_PRIVATE_KEY")],
```

Comment:
```
yarn hardhat keystore set DEPLOYER_PRIVATE_KEY
```

We'd need to abstract this to the user for `yarn generate` etc.

Also: for the keystore you need a 8 character password. Unless you use the `--dev` flag, which doesn't ask for a password at all https://hardhat.org/docs/guides/configuration-variables#improving-ux-when-using-keystore-values-during-the-dev-process